### PR TITLE
fix: allow typescript-eslint@8 as dependency

### LINKED ISSUE #148: @typescript-eslint/utils dependency should support typescript-eslint@8
## Overview

👋 Hi! [typescript-eslint v8 is released](https://typescript-eslint.io/blog/announcing-typescript-eslint-v8). `eslint-plugin-solid` only supports `^7.13.1` at the moment:

https://github.com/solidjs-community/eslint-plugin-solid/blob/493e5d386e86b0300abd4b03199897a3deb3929f/package.json#L41

Would you be open to a PR that adds `|| ^8.0.0`?

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Fixes #148.

~Sending as a draft for reference in case it's useful. Cheers! 😄~ Now that typescript-eslint@8 is released, this is ready for review + merge IMO.

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,1606 +1,3282 @@
-lockfileVersion: '6.0'
+lockfileVersion: '9.0'
```
Comment (JoshuaKGoldberg) [inline pnpm-lock.yaml:1]: Which version of `pnpm` does this repo require? I'm on a newer one, so this lockfile got pretty significantly changed here.

**Review by pauliesnug — APPROVED** [review]

File: pnpm-lock.yaml (hunk 1 above)
Comment (eagerestwolf) [reply pnpm-lock.yaml:1]: Currently, there is no `packageManager` field, so there is no predefined version to use. When you used a newer version of `pnpm`, it probably added a `packageManager` field to the package.json

Comment (JoshuaKGoldberg) [reply pnpm-lock.yaml:1]: Yup, I was hoping to be told what version is preferred so I can just go with that - instead of changing a bunch of things.

**Comment by joshwilsonvu** [comment]
Thanks so much for this!

**Pull request merged**